### ISSUE TITLE
[New-TestResources] ServicePrincipal is not assigned as a resource group owner

### ISSUE THREAD

**Issue by pakrym** [Azure/azure-sdk-tools#2399]
```
.\eng\common\TestResources\New-TestResources.ps1 tables -basename pakrymtables2
4:52:05 PM - Detected .NET repository. Defaulting OutFile to true. Test environment settings would be stored into the file so you don't need to set environment variables manually.
4:52:05 PM - Using subscription 'Azure SDK Developer Playground (faa080af-c1d8-40ad-9cce-e1a450ca5b57)'
WARNING: Any clean-up scripts running against subscription 'faa080af-c1d8-40ad-9cce-e1a450ca5b57' may delete resource group 'rg-pakrymtables2' after 120 hours.
4:52:07 PM - Creating resource group 'rg-pakrymtables2' in location 'westus2'
4:52:10 PM - TestApplicationId was not specified; loading cached service principal '58fa600d-c62f-4e43-8dd7-7efb6d9132fa'
4:52:10 PM - Deployment template D:\github\azure\net\sdk\tables\test-resources.json to resource group rg-pakrymtables2

```

```
az role assignment list --resource-group rg-pakrymtables2
[
  {
    "canDelegate": null,
    "condition": null,
    "conditionVersion": null,
    "description": null,
    "id": "/subscriptions/faa080af-c1d8-40ad-9cce-e1a450ca5b57/resourcegroups/rg-pakrymtables2/providers/Microsoft.Authorization/roleAssignments/470ac095-70b3-5294-b3ec-15f52a5cec8b",
    "name": "470ac095-70b3-5294-b3ec-15f52a5cec8b",
    "principalId": "ef082e08-2579-436b-83e4-19183223dc5e",
    "principalName": "http://test-resources-pakrymtablesefcd.microsoft.com",
    "principalType": "ServicePrincipal",
    "resourceGroup": "rg-pakrymtables2",
    "roleDefinitionId": "/subscriptions/faa080af-c1d8-40ad-9cce-e1a450ca5b57/providers/Microsoft.Authorization/roleDefinitions/0a9a7e1f-b9d0-4cc4-a60d-0319b160aaa3",
    "roleDefinitionName": "Storage Table Data Contributor",
    "scope": "/subscriptions/faa080af-c1d8-40ad-9cce-e1a450ca5b57/resourcegroups/rg-pakrymtables2",
    "type": "Microsoft.Authorization/roleAssignments"
  }
]
```

/cc @jsquire @benbp 

This breaks the .NET resource group expiration logic.

**Comment by jsquire** [comment]
This doesn't make sense to me:

- If it's a newly created principal, it is granted ownership of the RG here:  _([src](https://github.com/Azure/azure-sdk-for-net/blob/main/eng/common/TestResources/New-TestResources.ps1#L540))_ 
- If it's an existing principal, it is granted ownership of the RG here:  _([src](https://github.com/Azure/azure-sdk-for-net/blob/main/eng/common/TestResources/New-TestResources.ps1#L590))_

We do treat the latter assignment as a warning rather than an exception, as there are some scenarios where the existing principal has subscription-level rights and we don't care about the assignment - but that'll raise a warning log message.

**Comment by pakrym** [comment]
The cached principal might've been created with a different resource group name even for a completely different service library.

**Comment by jsquire** [comment]
Fixed by #2412
